### ISSUE TITLE
permutation without repetition with selected object r from n things

### ISSUE THREAD

**Issue by ramAdam** [dpaukov/combinatoricslib#20]
how do you do permutation without repetition with selected objects r 
P(n , r) = n! / (n - r)!

where n is the number of things to choose from, and we choose r of them
Is there any API doc available from the library?

**Comment by dpaukov** [comment]
If you need to generate the permutations you can look at the documentation  [here](https://github.com/dpaukov/combinatoricslib#1-simple-permutations). More details can be found in the [permutation iterator class](https://github.com/dpaukov/combinatoricslib/blob/master/src/main/java/org/paukov/combinatorics/permutations/PermutationIterator.java). Method `next()` produces next permutation every time it is called during iterating over all permutations.

**Comment by ramAdam** [comment]
anyone looking for the answer to the question above may find the following answer useful

```
/*permutation without repetition*/
   ICombinatoricsVector<String> originalVector = Factory.createVector(new String[] { "apple", "orange" });
   Generator<String> gen = Factory.createPermutationWithRepetitionGenerator(originalVector, 3);
   
  for (ICombinatoricsVector<String> perm : gen){
      if( !perm.hasDuplicate())      //filters all the ones without duplicates
           System.out.println( perm );
   }
```

**Comment by dpaukov** [comment]
@ramAdam your example above will return nothing, as you try to generate 3-permutations of the 2 elements set of `("apple", "orange")`. All the permutations will have duplicates in the example. Permutations' length shouldn't be greater than the cardinality of the set.

**Comment by ramAdam** [comment]
@dpaukov you are correct, i copy pasted the code from docs, forgot to edit elements of the set . I edited the post again, could you please check now. Thanks
